ground/src/main/java/com/google/android/ground/ui/editsubmission/PhotoTaskViewModel.kt

```diff
@@ -24,16 +25,53 @@ import com.google.android.ground.model.submission.TextTaskData.Companion.fromStr
 import com.google.android.ground.model.task.Task
 import com.google.android.ground.persistence.remote.firestore.FirestoreStorageManager.Companion.getRemoteMediaPath
 import com.google.android.ground.repository.UserMediaRepository
+import com.google.android.ground.rx.annotations.Cold
 import com.google.android.ground.rx.annotations.Hot
+import com.google.android.ground.system.PermissionsManager
+import com.google.android.ground.ui.util.BitmapUtil
+import io.reactivex.Completable
+import io.reactivex.Observable
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.PublishSubject
+import io.reactivex.subjects.Subject
 import java.io.File
 import java.io.IOException
 import javax.inject.Inject
 import timber.log.Timber
 
 class PhotoTaskViewModel
 @Inject
-constructor(private val userMediaRepository: UserMediaRepository, resources: Resources) :
-  AbstractTaskViewModel(resources) {
+constructor(
+  private val userMediaRepository: UserMediaRepository,
+  private val permissionsManager: PermissionsManager,
+  private val bitmapUtil: BitmapUtil,
+  resources: Resources
+) : AbstractTaskViewModel(resources) {
+
+  // TODO(jsunde): This logic is currently duplicated here and in EditSubmissionViewModel.
+  //  If it's here it can easily be used for the DataCollectionFragment, but is slightly more
+  //  complicated to reuse from the EditSubmissionFragment. I should explore some more to see if
+  //  it's possible to remove this logic from the EditSubmissionViewModel
```

Comment:
@gino-m Do you have any suggestions for how to dedupe some of this logic? I was hoping to be able to reuse the photo handling logic here and in the EditSubmissionFragment, but I was having a hard time seeing how I would do that when the individual ViewModels are not restored from the SavedInstanceState so I'm not sure how we would keep track of the task that was waiting for the photo in the EditSubmissionFragment.

Reply:
Rather than managing view models in `getTaskViewModel`, why not binding them 1:1 to fragments and let Jetpack handle them for you as we do elsewhere? I believe the ViewModels should then survive the photo intent.

Reply:
Actually, I believe the app can be destroyed while in the photo intent, so we'd need to manage the persistent state in `SaveInstanceState`.

Reply:
Synced offline and we decided that we don't need to address this at this time since the EditSubmissionFragment will likely end up getting removed in favor of an edit mode for the data collection flow.